internal/cmd/gocdk/_static/demo/blob.bucket/demo_blob_bucket.go

```diff
@@ -71,101 +90,193 @@ const blobBucketBaseTemplate = `
     <li><a href="./list">List</a> the contents of the bucket</li>
     <li><a href="./view">View</a> the contents of a specific blob in the bucket</li>
     <li><a href="./write">Write</a> a new blob into the bucket</li>
-  </ul>
+</ul>
+{{if .Err}}
+  <p><strong>{{.Err}}</strong></p>
+{{end}}`
+
+	blobBucketTemplateSuffix = `
 </body>
-</html>
-`
+</html>`
 
-func blobBucketBaseHandler(w http.ResponseWriter, req *http.Request) {
-	tmpl := template.Must(template.New("base").Parse(blobBucketBaseTemplate))
-	if err := tmpl.Execute(w, bucketURL); err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-	}
-}
+	// Input: *blobBucketBaseData.
+	blobBucketBaseTemplate = blobBucketTemplatePrefix + blobBucketTemplateSuffix
 
-// Input: []*blob.ListObject.
-const blobBucketListTemplate = `
-<!DOCTYPE html>
-<html>
-<head>
-  <meta charset="UTF-8">
-  <title>blob.Bucket demo</title>
-</head>
-<body>
-  {{range .}}
+	// Input: *blobBucketListData.
+	blobBucketListTemplate = blobBucketTemplatePrefix + `
+  {{range .ListObjects}}
     <div>
       {{if .IsDir}}
         <a href="./list?prefix={{ .Key }}">{{ .Key }}</a>
       {{else}}
         <a href="./view?key={{ .Key }}">{{ .Key }}</a>
       {{end}}
     </div>
+  {{end}}` + blobBucketTemplateSuffix
+
+	// Input: *blobBucketViewData.
+	blobBucketViewTemplate = blobBucketTemplatePrefix + `
+  {{if .ListObjects}}
+    <form>
+    Choose a blob to view:
+    <select name="key" label="choose a blob to view">
```

Comment:
Unfortunately, I don't think this will work, as `label` is an element but not a valid attribute. 

You can either give the `input` element an `id` attribute and have a `label` element with a `for` attribute with a value that is the same as the `id` value, or you can nest the input inside of the label. [MDN reference](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label)

Reply:
Ah, done, here and in `secrets` demo.